### PR TITLE
Fix get_integer_time precision and zero handling

### LINKED ISSUE #468: Improve get_integer_time: handle zero timedelta and avoid float precision issues
## Problem
The current implementation of `get_integer_time` in `neural_lam/utils.py` has two potential issues:

### 1. Ambiguous behavior for zero timedelta
For `timedelta(0)`, the function returns:
```
(0, "weeks")
```
This occurs because `0 % unit == 0` for all units, and the function returns the first match ("weeks").
While mathematically valid, this behavior may be unintuitive, since zero has no inherent scale and the choice of "weeks" is arbitrary.

### 2. Floating-point precision risk
The function relies on:
```python
total_seconds = tdelta.total_seconds()
```
and performs modulo operations with float-derived values.
Floating-point arithmetic can introduce precision inconsistencies in edge cases. For example:
```python
get_integer_time(timedelta(days=0.001))  # returns (1, 'unknown') instead of (86400, 'milliseconds')
```

---

## Steps to Reproduce
```python
from datetime import timedelta
from neural_lam.utils import get_integer_time

print(get_integer_time(timedelta(0)))
```

### Current Output
```
(0, "weeks")
```

---

## Expected Behavior
For `timedelta(0)`, returning:
```
(0, "seconds")
```
is more intuitive and avoids arbitrary unit selection.

---

### 1. Handle zero explicitly
```python
if total_microseconds == 0:
    return 0, "seconds"
```

### 2. Use integer microseconds for computation
Compute total microseconds directly from `timedelta`'s integer fields instead of using `total_seconds()`:
```python
total_microseconds = (
    tdelta.days * 86400_000000
    + tdelta.seconds * 1_000000
    + tdelta.microseconds
)
```
Python guarantees `.days`, `.seconds`, and `.microseconds` are integers, so this avoids floating-point precision issues entirely.

### 3. Maintain unit definitions in integer microseconds
```python
units = {
    "weeks":        604800_000000,
    "days":          86400_000000,
    "hours":          3600_000000,
    "minutes":          60_000000,
    "seconds":           1_000000,
    "milliseconds":          1_000,
    "microseconds":              1,
}
```

---

## Benefits
- Improves clarity for zero timedelta handling
- Eliminates floating-point precision issues
- Ensures consistent and predictable behavior
- Preserves existing `dict` structure with minimal changes

---

## Testing
The behavior has been verified with unit tests covering:
- Standard cases (days, hours)
- Unit conversion (milliseconds → seconds)
- Edge cases:
  - zero timedelta
  - negative timedelta

All tests pass after applying the proposed changes.

### PR DESCRIPTION
**Pull request by Saptami191**
## Describe your changes

Fix get_integer_time precision issues and zero timedelta handling.

Uses integer microseconds instead of float computation to avoid rounding errors and ensure correct unit detection.

No additional dependencies.

## Issue Link

Closes #468 

## Type of change

- [x] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📖 Documentation (Addition or improvements to documentation)

## Checklist before requesting a review

- [x] My branch is up-to-date with the target branch - if not update your fork with the changes from the target branch (use `pull` with `--rebase` option if possible).
- [x] I have performed a self-review of my code
- [x] For any new/modified functions/classes I have added docstrings that clearly describe its purpose, expected inputs and returned values
- [x] I have placed in-line comments to clarify the intent of any hard-to-understand passages of my code
- [ ] I have updated the [README](README.MD) to cover introduced code changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have given the PR a name that clearly describes the change, written in imperative form
- [ ] I have requested a reviewer and an assignee

## Checklist for reviewers

Each PR comes with its own improvements and flaws. The reviewer should check the following:
- [ ] the code is readable
- [ ] the code is well tested
- [ ] the code is documented (including return types and parameters)
- [ ] the code is easy to maintain

## Author checklist after completed review

- [x] I have added a line to the CHANGELOG describing this change, in a section
  reflecting type of change:
  - fixes

## Checklist for assignee

- [ ] PR is up to date with the base branch
- [ ] the tests pass
- [ ] (if the PR is not just maintenance/bugfix) the PR is assigned to the next milestone. If it is not, propose it for a future milestone.
- [ ] author has added an entry to the changelog (and designated the change as *added*, *changed*, *fixed* or *maintenance*)
- Once the PR is ready to be merged, squash commits and merge the PR.

### CODE REVIEW

**Comment by Saptami191** [comment]
@joeloskarsson Recreated this PR after fixing earlier issues with the template and workflow. The code, tests, and linting are now clean and aligned with project guidelines. Would appreciate a review.

File: neural_lam/utils.py [hunk 1]
```diff
@@ -601,22 +601,31 @@ def get_integer_time(tdelta) -> tuple[int, str]:
         >>> get_integer_time(timedelta(milliseconds=1000))
         (1, 'seconds')
         >>> get_integer_time(timedelta(days=0.001))
-        (1, 'unknown')
+        (86400, 'milliseconds')
+        >>> get_integer_time(timedelta(0))
+        (0, 'seconds')
     """
-    total_seconds = tdelta.total_seconds()
+    total_microseconds = (
```
Comment (kshirajahere) [inline neural_lam/utils.py:665]: The new implementation is fine, but this helper’s contract is still inconsistent across the tree: `compute_standardization_stats.py` still calls `get_integer_time(step_length.total_seconds())`, i.e. with a float rather than a `timedelta`. I think we should either update that caller in the same slice or explicitly tighten/document the contract here.

File: tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+def test_hours():
+    assert get_integer_time(timedelta(hours=5)) == (5, "hours")
+
+
+def test_zero():
+    assert get_integer_time(timedelta(0)) == (0, "seconds")
+
+
+def test_milliseconds():
```
Comment (kshirajahere) [inline tests/test_utils.py]: Since this PR is specifically about float-precision avoidance, I think it would be good to add one direct regression test for the issue-body example (`timedelta(days=0.001)` -> `(86400, "milliseconds")`) here as well, rather than leaving that behavior covered only by the doctest.

File: CHANGELOG.md [hunk 3]
```diff
@@ -32,6 +32,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Fix typo in `ar_model.py` that causes `AttributeError` during evaluation [\#204](https://github.com/mllam/neural-lam/pull/204) @ritinikhil
 
+- Fix `get_integer_time` to avoid floating-point precision issues and correctly handle zero timedelta [#469](https://github.com/mllam/neural-lam/pull/469) @Saptami191
```
Comment (kshirajahere) [inline CHANGELOG.md]: Shud point to #494 instead of #469

**Review by kshirajahere — COMMENTED** [review]
I think the integer-microseconds direction is the right fix, but I think there is still few thing worth addressing before merge.

File: CHANGELOG.md (hunk 3 above)
Comment (Saptami191) [reply CHANGELOG.md]: Thanks for the review! I hve addressed all three points.

File: tests/test_utils.py (hunk 2 above)
Comment (observingClouds) [reply tests/test_utils.py]: This is actually already tested in the docstring.

File: tests/test_utils.py
Comment (observingClouds) [inline tests/test_utils.py:1]: All these newly added tests are already covered by the doctest. No need to replicate them here. If you think a case is missing in the docstring, please add them there.

**Review by observingClouds — CHANGES_REQUESTED** [review]
Thanks for this PR. I left a few comments and suggest a few changes.

File: tests/test_utils.py
Comment (Saptami191) [reply tests/test_utils.py:1]: @observingClouds thanks for the review! yeah all are tested in docstring. i will remove this test file .actually it was part of proving the fix is correct.
Should I also add a doctest for negative timedelta
 >>> get_integer_time(timedelta(days=-7))
    (-1, 'weeks')  

it will represent a duration in the past 7 days ago.
 or is that out of scope for this PR?

**Comment by Saptami191** [comment]
@observingClouds Should `get_integer_time` explicitly support negative timedeltas (e.g., `timedelta(days=-7) -> (-1, "weeks")`), or is that out of scope for this PR?

File: tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    assert get_integer_time(timedelta(milliseconds=1000)) == (1, "seconds")
+
+
+def test_negative():
+    assert get_integer_time(timedelta(days=-7)) == (-1, "weeks")
```
Comment (observingClouds) [inline tests/test_utils.py]: Yes, please add this test to the doctest and then all is covered by the doctest and we can remove all tests here

**Review by observingClouds — CHANGES_REQUESTED** [review]

**Comment by Saptami191** [comment]
@observingClouds Hii! Requested changes are done and all tests have been passed

**Review by sadamov — APPROVED** [review]
@observingClouds @Saptami191 I just rebased this PR onto main. Feel free to have a look and merge if you agree.
